test/unit/app/controllers/incoming-transactions-test.js

```diff
@@ -139,28 +182,443 @@ describe('IncomingTransactionsController', function () {
     })
   })
 
-  describe('#start', function () {
-    it('should set up a listener for the latest block', function () {
+  describe('update events', function () {
+    it('should set up a listener for the latest block', async function () {
       const incomingTransactionsController = new IncomingTransactionsController({
         blockTracker: getMockBlockTracker(),
         networkController: getMockNetworkController(),
         preferencesController: getMockPreferencesController(),
         initState: {},
       })
-      sinon.spy(incomingTransactionsController, '_update')
 
       incomingTransactionsController.start()
 
       assert(incomingTransactionsController.blockTracker.addListener.calledOnce)
       assert.equal(incomingTransactionsController.blockTracker.addListener.getCall(0).args[0], 'latest')
-      const blockTrackerListenerCallback = incomingTransactionsController.blockTracker.addListener.getCall(0).args[1]
-      assert.equal(incomingTransactionsController._update.callCount, 0)
-      blockTrackerListenerCallback('0xabc')
-      assert.equal(incomingTransactionsController._update.callCount, 1)
-      assert.deepEqual(incomingTransactionsController._update.getCall(0).args[0], {
-        address: '0x0101',
-        newBlockNumberDec: 2748,
+    })
+
+    it('should update upon latest block when started and on supported network', async function () {
```

Comment:
The phrase "supported network" appears multiple times across the new tests, but its meaning is unclear. In a lot of cases, the behavior of the controller appears to be the same whether the network is supported or not.

A comment somewhere in this file (and maybe the controller implementation itself) explaining what it means for a network to be supported would be helpful.

Reply:
Good point - I will add a description somewhere. Basically it was meant to mean "the networks supported by this controller", i.e. the ones that we can fetch incoming transactions for.

Functionally you can see the difference on this line: https://github.com/MetaMask/metamask-extension/blob/develop/app/scripts/controllers/incoming-transactions.js#L180
The function will bail there for any networks not in our network map, thus not actually retrieving any incoming transactions.

It does still update state for these cases, but only to track the "last block fetched". Despite the fact that it _hasn't fetched anything_ in practice. This behaviour is removed entirely in a subsequent PR - I will make it such that state is not updated at all in these cases.

Reply:
I decided to tackle this in a separate PR instead: https://github.com/MetaMask/metamask-extension/pull/9759